### PR TITLE
Authentication ticket #8 and #9

### PR DESCRIPTION
**Pull request by cecerodrgz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,5 +38,9 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "cors": "^2.8.5",
+    "express": "^4.18.2"
```
Comment (jpremo) [inline package.json:44]: We shouldn't need these packages on the frontend. Can you remove them or explain why they're necessary.

File: src/App.js [hunk 2]
```diff
@@ -1,30 +1,40 @@
-import React, { useState, useEffect } from 'react';
-import { Routes, Route } from 'react-router-dom';
-import { getAllData } from './util/index';
-import { HomePage } from './components/HomePage/HomePage';
-import { LoginPage } from './components/LoginPage/LoginPage';
-import ProtectedRoute from './components/ProtectedRoute/ProtectedRoute';
```
Comment (jpremo) [inline src/App.js:6]: This all needs to still be imported (other than the duplicate react import. I think this got deleted during the merge conflict. In this case you probably wanted to accept both changes and then delete the double import.

File: src/App.js [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
+import React, { useState, useEffect } from "react";
+import { getAllData } from "./util/index";
 
-const URL = 'http://localhost:8000/api/v1/';
```
Comment (jpremo) [inline src/App.js:8]: This still needs to be included. Please add it back in.

File: src/App.js [hunk 4]
```diff
@@ -1,30 +1,40 @@
...
 
-const App = () => {
-
-const [message, setMessage] = useState(''); 
+function App() {
+  const [, setMessage] = useState("");
```
Comment (jpremo) [inline src/App.js:6]: message should be defined here.

File: src/App.js [hunk 5]
```diff
@@ -1,30 +1,40 @@
...
+          <input/>
+          <br />
+        </label>
+        <div>
+          <button type="submit"> Submit </button>
+        </div>
+      </form>
+</div>
     <>
       <h1>{message}</h1>
```
Comment (jpremo) [inline src/App.js:39]: This fragment needs to be at the top of the return. We're getting an error right now.

File: src/util/Authentication.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  if (user !== ""){
+    alert ("Welcome back!" + user);
+  } else {
+    user = prompt("Please enter your username", " ");
+    if(user !== "" && user != null){
+      setCookie("username", user, 365);
+    }
+  }
+  
+}
```
Comment (jpremo) [inline src/util/Authentication.js:36]: This function can be deleted since it's more of an example function.

File: src/util/Authentication.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    if(element.indexOf(cName) === 0){
+        result = element.substring(cName.length + 1)
+    }
+  })
+  return result;
+}
+
+function deleteCookie(cName){
+    setCookie(cName, null, null)
+}
```
Comment (jpremo) [inline src/util/Authentication.js:23]: All three functions work great. Nice job. Can you export each of these from the file?

File: src/util/fetch.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+fetch("url", {
+    Method: "POST",
+    Headers: {
+        Authorization: "cookie",
+        Accept: "aplication.json",
+        "Content-Type": "application/json"
+    },
+    Body: body,
+    Cache: "default"
+})
```
Comment (jpremo) [inline src/util/fetch.js:10]: For this we need to define a function. It should look something like this:
```
export const fetchAPIData = async(url, method, body, headers) => {
    //get the cookie here and pull off the token value stored in the cookie.
    const data = await fetch(url, {
        method,
        body: JSON.stringify(body),
        Headers: {
            Authorization: 'Bearer ${insertCookieToken here}',
            'Content-Type': 'application/json',
            ...headers,
        },
    })
    //process response to json
}
```

File: src/util/Authentication.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  const cDecoded = decodeURIComponent(document.cookie);
+  const cArray = cDecoded.split( "; ");
+  let result = null;
+
+  cArray.forEach(element => {
+    if(element.indexOf(cName) === 0){
+        result = element.substring(cName.length + 1)
+    }
+  })
+  return result;
```
Comment (jpremo) [inline src/util/Authentication.js:18]: Before returning this can you convert from JSON to an object with JSON.parse?

**Review by jpremo — CHANGES_REQUESTED** [review]
The cookie functions work well. Nice job on that! It looks like the merge conflict had a few issues. Some code in App.js got deleted and is causing some errors. It's always good to test after resolving merge conflicts to make sure everything still works.

Let me know if you have any questions on any of the comments.

File: package.json (hunk 1 above)
Comment (cecerodrgz) [reply package.json:44]: These were added while I was trying some stuff to make the login, I forgot that they were added.

File: src/App.js (hunk 3 above)
Comment (cecerodrgz) [reply src/App.js:8]: This was only deleted on the PR, I still have it on my local branch. Sorry about that!

File: package.json (hunk 1 above)
Comment (jpremo) [reply package.json:44]: These still need to be deleted.

**Review by jpremo — CHANGES_REQUESTED** [review]

File: src/util/Authentication.js (hunk 7 above)
Comment (jpremo) [reply src/util/Authentication.js]: These still need to be exported from the file using the **export** keyword.

**Review by jpremo — APPROVED** [review]

**Pull request merged**